core/dbt/include/global_project/macros/materializations/models/incremental/incremental.sql

```diff
@@ -50,9 +50,14 @@
 
     {#-- Get the incremental_strategy, the macro to use for the strategy, and build the sql --#}
     {% set incremental_strategy = config.get('incremental_strategy') or 'default' %}
-    {% set incremental_predicates = config.get('incremental_predicates', none) %}
+    {% set incremental_predicates = config.get('incremental_predicates', []) %}
+    {% set predicates = [] %}
+    {% for predicate in incremental_predicates %}
+      {#-- Wrap predicates in parentheses to ensure correct evaluation when later anded --#}
+      {% do predicates.append('(' ~ predicate ~ ')') %}
```

Comment:
Oh! I see what you mean, since predicates could have their own nested conditional logic (and/or):
```sql
(predicate_a) and (predicate_b) and (predicate_c)
```
Would it make more sense to do that wrapping here, or within the templating logic in `get_merge_sql`?

Reply:
I concluded here was best, as it avoids the need to replicate this logic into every `get_x_sql` which might use the predicates. I'll defer to you though!